webapp/src/components/modals/create_issue/index.tsx

```diff
@@ -0,0 +1,280 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {useDispatch, useSelector} from 'react-redux';
+import {Modal} from 'react-bootstrap';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {getPost} from 'mattermost-redux/selectors/entities/posts';
+
+import {getErrorMessage} from 'src/utils/user_utils';
+import GitlabLabelSelector from 'src/components/gitlab_label_selector';
+import GitlabAssigneeSelector from 'src/components/gitlab_assignee_selector';
+import GitlabMilestoneSelector from 'src/components/gitlab_milestone_selector';
+import GitlabProjectSelector from 'src/components/gitlab_project_selector';
+import Validator from 'src/components/validator';
+import FormButton from 'src/components/form_button';
+import Input from 'src/components/input';
+import {id as pluginId} from 'src/manifest';
+import {closeCreateIssueModal, createIssue} from 'src/actions';
+import {GlobalState} from 'src/types/global_state';
+import {usePrevious} from 'src/hooks/use_previous';
+
+const MAX_TITLE_LENGTH = 255;
+
+type PropTypes = {
+    theme: Theme;
+};
+
+const CreateIssueModal = ({theme}: PropTypes) => {
+    const validator = useMemo(() => (new Validator()), []);
+    const [submitting, setSubmitting] = useState<boolean>(false);
+    const [error, setError] = useState<string | null>(null);
+    const [project, setProject] = useState<ProjectSelection | null>(null);
+    const [issueTitle, setIssueTitle] = useState<string>('');
+    const [issueDescription, setIssueDescription] = useState<string>('');
+    const [labels, setLabels] = useState<SelectionType[]>([]);
+    const [assignees, setAssignees] = useState<SelectionType[]>([]);
+    const [milestone, setMilestone] = useState<SelectionType | null>(null);
+    const [showErrors, setShowErrors] = useState<boolean>(false);
+    const [issueTitleValid, setIssueTitleValid] = useState<boolean>(true);
+
+    const {visible, post, channelId, title} = useSelector((state: GlobalState) => {
+        const {postId, title, channelId} = state[`plugins-${pluginId}` as plugin].createIssueModal;
+        
+        const post = postId ? getPost(state, postId) : null;
+        return {
+            visible: state[`plugins-${pluginId}` as plugin].isCreateIssueModalVisible,
+            post,
+            title,
+            channelId,
+        };
+    });
```

Comment:
This modal should be split into two components. One should be "should I render right now?" And if yes, it should render the `Modal` component with the other component nested inside. This allows React to clean up state and we don't end up with code like `handleClose` doing a bunch of cleanup.

Same for attach comment modal as well